clover.rb

```diff
@@ -510,4 +562,18 @@ def assets
     rodauth.require_authentication
     r.hash_branches("")
   end
+
+  # Validate response against OpenAPI schema
+  after do |res|
+    status, headers, body = res
+    next unless api? && status && headers && body
+    schema_validator = SCHEMA_ROUTER.build_schema_validator(request)
```

Comment:
This duplicates what the main route block is doing.  I'm not sure how expensive `SCHEMA_ROUTER.build_schema_validator` is, but it seems like setting it to an instance variable in the main route, and using `||=` here, would avoid duplicate code.

Reply:
I don't think it's too bad expense-wise, but I agree that memoizing it would be worth doing.

Reply:
I put in a rather brutal memoization. The brutality is there's an aspect of repeating myself. I played around with some of the conditionals a tiny bit, I was a bit puzzled how I could not assume `@schema_validator` set in some cases, rather than using `||=` at the last site. I did confirm that `@schema_validator` was set much of the time via print expressions, though.

Reply:
You cannot assume it will be present, because it isn't set until fairly late in the main route block, after `r.rodauth` and some `rodauth` methods are called.   Setting `@schema_validator` would probably have to be the first thing in the route block (for api routes at least) for you to rely on it being present in the after hook.